test/thumbnailer_test.cc

```diff
@@ -100,88 +96,117 @@ class WebPSamplesGenerator {
-
   const int pic_count = 10;
   const uint8_t transparency = 0xaf;
   const bool use_randomized = false;
-  auto test_pics = WebPSamplesGenerator(pic_count, transparency, use_randomized)
-                       .GeneratePics();
 
-  for (int i = 0; i < pic_count; ++i) {
-    thumbnailer.AddFrame(*test_pics[i].get(), i * 500);
-  }
   std::unique_ptr<WebPData, void (*)(WebPData*)> webp_data(new WebPData,
                                                            WebPDataDelete);
   WebPDataInit(webp_data.get());
-  thumbnailer.GenerateAnimation(webp_data.get());
+  bool ok = WebPTestGenerator(pic_count, transparency, use_randomized)
+                .GenerateWebPData(webp_data.get());
 
+  EXPECT_EQ(ok, true);
   EXPECT_LE(webp_data->size, kDefaultBudget);
   EXPECT_GT(webp_data->size, 0);
 }
 
 TEST(ThumbnailerTest, NoisyImageSolid) {
-  libwebp::Thumbnailer thumbnailer = libwebp::Thumbnailer();
-
   const int pic_count = 10;
   const uint8_t transparency = 0xff;
   const bool use_randomized = true;
-  auto test_pics = WebPSamplesGenerator(pic_count, transparency, use_randomized)
-                       .GeneratePics();
 
-  for (int i = 0; i < pic_count; ++i) {
-    thumbnailer.AddFrame(*test_pics[i].get(), i * 500);
-  }
   std::unique_ptr<WebPData, void (*)(WebPData*)> webp_data(new WebPData,
                                                            WebPDataDelete);
   WebPDataInit(webp_data.get());
-  thumbnailer.GenerateAnimation(webp_data.get());
+  bool ok = WebPTestGenerator(pic_count, transparency, use_randomized)
+                .GenerateWebPData(webp_data.get());
 
+  EXPECT_EQ(ok, true);
   EXPECT_LE(webp_data->size, kDefaultBudget);
   EXPECT_GT(webp_data->size, 0);
 }
 
 TEST(ThumbnailerTest, NoisyImageTransparent) {
-  libwebp::Thumbnailer thumbnailer = libwebp::Thumbnailer();
+  const int pic_count = 10;
+  const uint8_t transparency = 0xaf;
+  const bool use_randomized = true;
+
+  std::unique_ptr<WebPData, void (*)(WebPData*)> webp_data(new WebPData,
+                                                           WebPDataDelete);
+  WebPDataInit(webp_data.get());
+  bool ok = WebPTestGenerator(pic_count, transparency, use_randomized)
+                .GenerateWebPData(webp_data.get());
+
+  EXPECT_EQ(ok, true);
+  EXPECT_LE(webp_data->size, kDefaultBudget);
+  EXPECT_GT(webp_data->size, 0);
+}
+
+TEST(ThumbnailerPSNRTest, NoisyImageSolid) {
+  const int pic_count = 10;
+  const uint8_t transparency = 0xff;
+  const bool use_randomized = true;
+
+  std::unique_ptr<WebPData, void (*)(WebPData*)> webp_data(new WebPData,
+                                                           WebPDataDelete);
+  WebPDataInit(webp_data.get());
+  bool ok = WebPTestGenerator(pic_count, transparency, use_randomized)
+                .GenerateWebPData(webp_data.get());
+
+  EXPECT_EQ(ok, true);
+  EXPECT_LE(webp_data->size, kDefaultBudget);
+  EXPECT_GT(webp_data->size, 0);
+}
 
+TEST(ThumbnailerPSNRTest, NoisyImageTransparent) {
```

Comment:
To improve further, you might want to take a look at [parameterized GoogleTest cases](http://sandordargo.com/blog/2019/04/24/parameterized-testing-with-gtest) because the only thing that seems to change between the test cases here are `pic_count`, `transparency` and `use_randomized`. The article also gives alternatives (fixtures, loops...)

Reply:
Tests will be left for the next PR.